src/ctypes/ldouble_stubs.c

```diff
@@ -453,23 +460,36 @@ static intnat ldouble_complex_hash(value v) {
 
 static void ldouble_complex_serialize(value v, uintnat *wsize_32, uintnat *wsize_64) {
   long double re,im;
-  long double _Complex *p = Data_custom_val(v);
+  long double _Complex c;
+  void * p = Data_custom_val(v);
+#if defined(__GNUC__) && __GNUC__  == 6 && __GNUC_MINOR__ == 4
+  /* workaround gcc bug. gcc tries to inline the memcpy calls, but
+   * fails with an internal compiler error. I've observed this error
+   * only under Alpine Linux, other distros have already imported a
+   * patch from upstream.
+   */
+  void *(*volatile mymemcpy)(void*,const void*,size_t) = memcpy;
+  mymemcpy(&c, p, sizeof(c));
+#else
+  memcpy(&c, p, sizeof(c));
+#endif
```

Comment:
Could we ask the Alpine people to compile using `-fno-builtin` instead of fixing the GCC problem here?

Reply:
Okay.  Since the fix is constrained to a specific GCC release, it should at least be clear when we can safely remove it.